core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -183,6 +189,10 @@ protected void subscribe0(EventExecutor executor, SubscriptionOption[] options)
                 } else {
                     executor.execute(() -> collectAll(collectingFuture, executor, withPooledObjects, false));
                 }
+            } else {
+                // We don't need to invoke onSubscribe() for NoopSubscriber.
+                invokedOnSubscribe = true;
```

Comment:
How about moving this up right after
`if (subscriptionUpdater.compareAndSet(this, null, subscription)) {`?